### PR TITLE
[CI] Select tests per label.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Few changes here:

1. We separate the platform labels and the tests labels.
2. We remove most if not all the implementation of IsIncluded in Jenkins.
3. We move from a string to an enum value per test.

### CODE REVIEW

**Review by chamons — APPROVED** [review]
LGTM

**Review by dalexsoto — APPROVED** [review]
Less strings is always good 👍

File: tests/xharness/TestLabel.cs [hunk 1]
```diff
@@ -18,85 +15,118 @@ public LabelAttribute (string label)
...
+	public enum TestLabel : Int64 {
 		[Label ("none")]
 		None = 0,
 		[Label ("bcl")]
 		Bcl = 1 << 1,
-		[Label ("bcl-nunit")]
-		BclNUnit = 1 << 2,
-		[Label ("bcl-xunit")]
-		BclXUnit = 1 << 3,
+		[Label("bgen")]
```
Comment (rolfbjarne) [inline tests/xharness/TestLabel.cs:23]:
```suggestion
		[Label ("bgen")]
```

File: tests/xharness/TestLabel.cs [hunk 2]
```diff
@@ -18,85 +15,118 @@ public LabelAttribute (string label)
...
+		Bgen = 1 << 2,
+		[Label ("binding")]
+		Binding = 1 << 3,
+		[Label ("bindings-framework")]
+		BindingFramework = 1 << 4,
+		[Label("bindings-xcframework")]
```
Comment (rolfbjarne) [inline tests/xharness/TestLabel.cs:29]:
```suggestion
		[Label ("bindings-xcframework")]
```

File: tests/xharness/Jenkins/TestSelector.cs [hunk 3]
```diff
@@ -33,14 +35,36 @@ public void SetEnabled (TestLabel label, bool enable)
...
+			} else {
+				platform &= ~label;
+			}
+		}
+
 		public void SetEnabled (string label, bool value)
 		{
-			var testLabel = label.GetLabel ();
-			SetEnabled (testLabel, value);
+			// there are two possible cases, either we are setting a test label OR a
```
Comment (tj-devel709) [inline tests/xharness/Jenkins/TestSelector.cs:49]:
```suggestion
			// there are two possible cases, either we are setting a test label OR a something else probably
```

File: tests/xharness/TestLabel.cs [hunk 4]
```diff
@@ -18,85 +15,118 @@ public LabelAttribute (string label)
...
-		tvOS = 1 << 15,
+		tvOS = 1 << 10,
 		[Label ("watchos")]
-		watchOS = 1 << 26,
-		[Label ("xtro")]
-		Xtro = 1 << 27,
+		watchOS = 1 << 11,
 		[Label ("all")]
-		All =  0xFFFFFFFF,
+		All =  0xFFFFFFFF,
```
Comment (tj-devel709) [inline tests/xharness/TestLabel.cs:107]:
```suggestion
		All =  0xFFFFFFFF,
```
nit, small extra space

**Review by tj-devel709 — APPROVED** [review]

**Pull request closed without merging**